coverage_point_calculator/src/lib.rs

```diff
@@ -0,0 +1,933 @@
+///
+/// ## Fields:
+/// - modeled_coverage_points
+///   - [HIP-74][modeled-coverage]
+///   - reduced cbrs radio coverage points [HIP-113][cbrs-experimental]
+///
+/// - assignment_multiplier
+///   - [HIP-103][oracle-boosting]
+///
+/// - rank
+///   - [HIP-105][hex-limits]
+///
+/// - hex_boost_multiplier  
+///   - must meet minimum subscriber thresholds [HIP-84][provider-boosting]
+///   - Wifi Location trust score >0.75 for boosted hex eligibility [HIP-93][wifi-aps]
+///
+/// - location_trust_score_multiplier
+///   - [HIP-98][qos-score]
+///   - states 30m requirement for boosted hexes [HIP-107][prevent-gaming]
+///   - increase Boosted hex restriction, 30m -> 50m [Pull Request][boosted-hex-restriction]
+///
+/// - speedtest_multiplier
+///   - [HIP-74][modeled-coverage]
+///   - added "Good" speedtest tier [HIP-98][qos-score]
+///     - latency is explicitly under limit in HIP https://github.com/helium/oracles/pull/737
+///
+/// ## Notable Conditions:
+/// - Location
+///   - If a Radio covers any boosted hexes, [LocationTrust] scores must meet distance requirements, or be degraded.
+///   - CBRS Radio's location is always trusted because of GPS.
+///
+/// - Speedtests
+///   - The latest 6 speedtests will be used.
+///   - There must be more than 2 speedtests.
+///
+/// - Covered Hexes
+///   - If a Radio is not [eligible_for_boosted_hexes], boost values are removed before calculations. [CoveredHexes::new_without_boosts]
+///
+/// ## References:
+/// [modeled-coverage]:        https://github.com/helium/HIP/blob/main/0074-mobile-poc-modeled-coverage-rewards.md#outdoor-radios
+/// [provider-boosting]:       https://github.com/helium/HIP/blob/main/0084-service-provider-hex-boosting.md
+/// [wifi-aps]:                https://github.com/helium/HIP/blob/main/0093-addition-of-wifi-aps-to-mobile-subdao.md
+/// [qos-score]:               https://github.com/helium/HIP/blob/main/0098-mobile-subdao-quality-of-service-requirements.md
+/// [oracle-boosting]:         https://github.com/helium/HIP/blob/main/0103-oracle-hex-boosting.md
+/// [hex-limits]:              https://github.com/helium/HIP/blob/main/0105-modification-of-mobile-subdao-hex-limits.md
+/// [prevent-gaming]:          https://github.com/helium/HIP/blob/main/0107-preventing-gaming-within-the-mobile-network.md
+/// [cbrs-experimental]:       https://github.com/helium/HIP/blob/main/0113-reward-cbrs-as-experimental.md
+/// [mobile-poc-blog]:         https://docs.helium.com/mobile/proof-of-coverage
+/// [boosted-hex-restriction]: https://github.com/helium/oracles/pull/808
+///
+use crate::{
+    hexes::{CoveredHex, CoveredHexes},
+    location::{LocationTrust, LocationTrustScores},
+    speedtest::Speedtest,
+};
+use coverage_map::{RankedCoverage, SignalLevel};
+use rust_decimal::{Decimal, RoundingStrategy};
+use rust_decimal_macros::dec;
+use speedtest::Speedtests;
+
+pub mod hexes;
+pub mod location;
+pub mod speedtest;
+
+pub type Result<T = ()> = std::result::Result<T, Error>;
```

Comment:
Move the error definition up here to be with the result

Reply:
Moved. They're not best friends.